app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/package.scala

```diff
@@ -167,6 +168,17 @@ package object controllers {
 
     def utr: Mapping[String] = text verifying utrConstraint
 
+    private val ninoConstraint: Constraint[String] = Constraint[String] { fieldValue: String =>
+      val formattedField = fieldValue.replace(" ", "")
```

Comment:
Can the fieldValue contain whitespace other than spaces that needs to be stripped, e.g. tabs? Maybe consider `.replaceAll("\\s", "")`